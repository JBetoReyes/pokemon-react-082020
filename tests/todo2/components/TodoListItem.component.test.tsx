import React from 'react';
import TodoListItem from '@components/hooks/Todo2/TodoListItem.component';
import { shallow } from 'enzyme';
import demoTodos2 from '../../fixtures/demoTodos2';

describe('TodoListItem', () => {
  const handleDelete = jest.fn();
  const handleTodo = jest.fn();
  const todoListItemJsx = (
    <TodoListItem
      todo={demoTodos2[0]}
      index={0}
      handleDelete={handleDelete}
      handleTodo={handleTodo}
    />
  );
  let wrapper = shallow(todoListItemJsx);
  beforeEach(() => {
    handleDelete.mockClear();
    handleTodo.mockClear();
    wrapper = shallow(todoListItemJsx);
  });
  test('Should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Should call the delete handler', () => {
    const deleteButton = wrapper.find('button');
    deleteButton.simulate('click');
    expect(handleDelete).toHaveBeenCalled();
    expect(handleDelete).toHaveBeenCalledWith(1);
  });
  test('Should call the handle toggle', () => {
    const p = wrapper.find('p');
    p.simulate('click');
    expect(handleTodo).toHaveBeenCalled();
    expect(handleTodo).toHaveBeenCalledWith(1);
  });
  test('Should have the correct label item', () => {
    const p = wrapper.find('p');
    expect(p.text()).toBe(`1. ${demoTodos2[0].description}`);
  });
  test('Should have the complete class', () => {
    wrapper = shallow(
      <TodoListItem
        todo={{ ...demoTodos2[0], done: true }}
        index={0}
        handleDelete={handleDelete}
        handleTodo={handleTodo}
      />
    );
    const p = wrapper.find('p');
    expect(p.hasClass('complete'));
  });
});
