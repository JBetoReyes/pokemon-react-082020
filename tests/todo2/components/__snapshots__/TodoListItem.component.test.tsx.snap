// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem Should match with snapshot 1`] = `
<Fragment>
  <li
    className="list-group-item"
  >
    <p
      className="m-5 "
      onClick={[Function]}
    >
      1. test todo 1
    </p>
    <button
      className="btn btn-outline-danger"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </li>
</Fragment>
`;
